projects/canopy/src/styles/variables/_variants.scss

```diff
@@ -8,6 +8,7 @@
   --generic-link-active-bg-color: var(--color-white);
   --generic-link-focus-color: var(--color-white);
   --generic-link-focus-bg-color: var(--color-black);
+  --generic-link-active-hover-color: var(--color-charcoal);
```

Comment:
Same as my comment on the sibling PR:

https://github.com/Legal-and-General/canopy/pull/351#discussion_r628215597

Reply:
I do have a small question on this one. If I get rid of the variables here, then I have to add the `@if @else` in the mixins.scss, and there will be some boilerplate code, as in master. Is that ok or did you have something else in mind?  :)

Reply:
can jump on a quick call to talk this over id needed :)

Reply:
Done like in the other PR and the button states now uses variables of colours and not the link defined ones. :)